controllers/users.js

```diff
@@ -75,6 +78,43 @@ const getUsers = async (req, res) => {
   }
 };
 
+/**
+ * Returns the usernames of inactive/idle users/members
+ *
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+
+const getIdleUsers = async (req, res) => {
+  try {
+    const allUsers = await userQuery.fetchUsers(req.query);
+    const onlyMembers = await members.fetchUsersWithRole(ROLES.MEMBER);
+    const taskParticipants = await tasks.fetchActiveTaskMembers();
+    const membersOnly = req.query["members-only"];
+
+    if (membersOnly === "true") {
+      const idleMembers = onlyMembers?.filter(({ id }) => !taskParticipants.has(id));
+      const idleMemberUserNames = idleMembers?.map((_member) => _member.username);
+
+      return res.json({
+        message: idleMemberUserNames.length ? "Idle members returned successfully!" : "No idle member found",
+        idleMemberUserNames,
+      });
+    }
+
+    const idleUsers = allUsers?.filter(({ id }) => !taskParticipants.has(id));
+    const idleUserUserNames = idleUsers?.map((_user) => _user.username);
+
+    return res.json({
+      message: idleUserUserNames.length ? "Idle users returned successfully!" : "No idle user found",
+      idleUserUserNames,
+    });
```

Comment:
we should not do it this way, instead, you can take a variable that is going to change based on the user query, and the n you can return that